### PR TITLE
docs: fix a bunch of broken item links

### LINKED ISSUE #1152: Fix Rustdoc links and add step to CI to check them.

### PR DESCRIPTION
**Pull request by dzvon**
Closes #1152

### Description

Correct some item links and fix some typo.

Some CI stuff will be added if I handled it. :D

### How was this PR tested?

`cargo doc --open` and check the result to see if it is right.

### CODE REVIEW

**Comment by dzvon** [comment]
CI job added.

**Comment by fulmicoton** [comment]
@dzvon Awesome thank you!

**Comment by dzvon** [comment]
Seems like some fmt checks failed. I'll take a look.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
         with:
           command: deny
           args: check Licenses # check that all of deps have an ok license.
+      - name: install cmake
```
Comment (fulmicoton) [inline .github/workflows/ci.yml:55]: why do we need cmake here?

**Comment by fulmicoton** [comment]
On your compute, you can run `make fmt` to format with the right conf.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dzvon) [reply .github/workflows/ci.yml:55]: `cargo doc` is failed on runner, and the message indicate that there is no cmake which is needed for crate `rdkafka-sys`. So I added a step to install cmake.

https://github.com/quickwit-oss/quickwit/runs/5531433392?check_suite_focus=true#step:8:158

Do you have a better idea?

**Comment by dzvon** [comment]
@fulmicoton Can you re-run this unit tests please? I want to check if [this test](https://github.com/quickwit-oss/quickwit/runs/5534899120?check_suite_focus=true#step:5:439) will always keep failing. Thank you.

I'm a little confused, because [it passed](https://github.com/dzvon/quickwit/runs/5535665255?check_suite_focus=true) successfully in my forked repo.

**Comment by fulmicoton** [comment]
@dzvon Great job!

**Pull request merged**